c++/include/orc/Vector.hh

```diff
@@ -113,6 +115,35 @@ namespace orc {
     DataBuffer<int64_t> length;
   };
 
+  struct StringDictionary {
+    StringDictionary(MemoryPool& pool);
+    DataBuffer<char> dictionaryBlob;
+
+    // Offset for each dictionary key entry.
+    DataBuffer<int64_t> dictionaryOffset;
+
+    void getValueByIndex(int64_t index, char*& valPtr, int64_t& length) {
```

Comment:
index as uint32_t?

Reply:
same as above, currently the RleDecoder only support int64_t, so we make it int64_t here as well.